rrdd/rrdd_server.ml

```diff
@@ -416,29 +417,51 @@ module Plugin = struct
 
 		(* Helper function to find plugin info from a uid. *)
 		let find ~(uid: P.uid) : plugin =
-			Mutex.execute registered_m
-				(fun () -> Hashtbl.find registered uid)
+			try
+				Mutex.execute registered_m (fun () -> Hashtbl.find registered uid)
+			with
+				Not_found ->
+					error "failed to find plugin %s" (P.string_of_uid uid);
+					raise Rrd_protocol.Read_error
```

Comment:
I see why `Read_error` is convenient, because you now use it in `get_payload`, but is it really appropriate at all call sites?